### PR TITLE
Completely Removes the CRAB 17

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Extension of the PR by CRIT #45739. Do note, I'm not apply the same logic he used for removal. 

## Why It's Good For The Game

CRAB Phone is at odds with the rest of our items. Its blatant, its loud, its in your face, and its annoying. Its a bad idea at its core, and while certain PRs could certainly make it more bearable (removing global sound), removal is the best due to the fact that no matter how much you dress it up, a bad idea is a bad idea. 

There are tons of good, creative ways to create a diversion as a traitor. If none of these work, a player could simply use a power sink. Due to the existence of the power sink, we do not need another blatant, round effecting device, all from one location. More often than not I see it used to meme or piss people off than to provide actual benefit, which usually is only available on nuke op rounds. 

## Changelog
:cl:
del: The entirety of CRAB-17 Protocol code, including sprites and sounds.
/:cl:

I think I was modifying older files so there'll probably be some conflicts I need to resolve

### CODE REVIEW

**Comment by ishitbyabullet** [comment]
> Due to the existence of the power sink, we do not need another blatant, round effecting device, all from one location.

I still don't get this argument. We have different guns, different melee weapons, different space suits, different ways to heal, and even different clothes. Why does there have to be a very strict limit to round affecting distractions that aren't lethal?

I can get the idea behind it being bad though. I like the meme value of it however not everyone does. A good thing to note is that the powersink lets you PREPARE where you can set it up. You can't do the same for the crab-17 though. This means one requires you to be stealthy and the other requires bare minimum stealth. That's an unique difference, just like the stechkin and revolver or esword and powerfist.

**Comment by wesoda25** [comment]
The point of all those guns and items are so that you have many different ways to go about creating whatever you want, in this case a diversion. Don't focus in on that being my only point for removalt tho

**Comment by ishitbyabullet** [comment]
> Don't focus in on that being my only point for removalt tho

Yeah, was about to mention another reason I remembered in an edit.

You said that it was really easy to tell what antag it was due to the centcom announcement. I thought of a way to fix this involving removing the announcement, turning it into a notification from ID/PDA, and making it a false alarm however that would require a lot of effort _so eeeeh_. It is true that this is the most easiest way to tell there are nuke ops or traitors since even the powersink, a guy in blatant chameleon armor, and player tc trades are less immediately obvious to the entire crew.

I have nothing to say for that though.

**Comment by wesoda25** [comment]
> > Don't focus in on that being my only point for removalt tho
> 
> Yeah, was about to mention another reason I remembered in an edit.
> 
> You said that it was really easy to tell what antag it was due to the centcom announcement. I thought of a way to fix this involving removing the announcement, turning it into a notification from ID/PDA, and making it a false alarm however that would require a lot of effort _so eeeeh_. It is true that this is the most easiest way to tell there are nuke ops or traitors since even the powersink, a guy in blatant chameleon armor, and player tc trades are less immediately obvious to the entire crew.
> 
> I have nothing to say for that though.

If a fucking little traitor item needs false alerts just to preserve the gamemode it should be removed. No item is that important.

**Comment by SkeletalElite** [comment]
> If a fucking little traitor item needs false alerts just to preserve the gamemode it should be removed. No item is that important.

Just make a crab 17 random event. It can still be forcefully bought and used by traitors/nukies but the gamemode isn't necessarily revealed by it happening.

**Comment by wesoda25** [comment]
bout to do something janky to see if it resolves the conflict

**Comment by wesoda25** [comment]
> > If a fucking little traitor item needs false alerts just to preserve the gamemode it should be removed. No item is that important.
> 
> Just make a crab 17 random event. It can still be forcefully bought and used by traitors/nukies but the gamemode isn't necessarily revealed by it happening.

why would I do that when I want it out of the game. I didn't even mention that argument in the PR body for a reason

**Comment by wesoda25** [comment]
well that was fucking wack but it would appear the merge conflict is indeed gone

**Comment by SynnGraffkin** [comment]
Crab Phone is a fun meme item, as long as the robot spawned isn't stupidly overbuffed to be indestructible.

**Comment by imsxz** [comment]
the fun of it is that it crowds a lot of people into a small area, and generally creates a really effective distraction. i've stripped sec officers next to it before because they simply didn't notice me amidst the sea of attack logs for people hitting it. people shooting it can be disarmed for free guns. great location to detonate bombs, release a gas, etc. 

Meanwhile, the powersink has a much different usage. For one, it doesn't really draw the crew to an area. It's more useful in other aspects, the main one being that it depowers all machinery and doors. I use it before releasing viruses, and people use it all the time to buy windows of time where they can easily b&e without sec being too concerned.  It explodes too, but that is rather rare.

tl;dr it does serve a unique function; it's a giant people magnet. i'm not sure there's any other function in the game to get that big of a group of people to an area other than HOP line on warops.

**Comment by wesoda25** [comment]
Thats basically the point I've been getting I've been getting at though. Its a boring and unimaginative way of getting a lot of people into one area. Traitors shouldn't have an easy item for accomplishing that.

**Comment by Qustinnus** [comment]
i remember when i originally PR'd this and people complained noone would go to the machine because noone would care about the money lmao.

anyways half of the original complaints were easy changes but instead the goal post has been moved so many times by Wesoda and his friends that I can't really do much with the feedback they ended up with, which is "I hate the item"

**Comment by ExcessiveUseOfCobblestone** [comment]
Do you guys actually care about giving the mode away and, if so, why does that care seem to stop at the CRAB item specifically? Is it because you cba to PR removals for the other items or something? Even then I haven't seen this complaint AT ALL on ANY CHANNEL unless we talk about this item specifically.

if you do care about giving the mode away, and you think the other channels for traitors to do this are acceptable, why not just have the CRAB use those same methods of communication instead as opposed to a full removal?

**Comment by ExcessiveUseOfCobblestone** [comment]
Also regardless of code changes to preserve the veil of the mode, there is nothing stopping a solo antagonist administratively from outting the mode via the communication tools offered to them such as radio/head request consoles/communication consoles.

**Comment by wesoda25** [comment]
> anyways half of the original complaints were easy changes but instead the goal post has been moved so many times by Wesoda and his friends that I can't really do much with the feedback they ended up with, which is "I hate the item"

I’ve already said it, I want it gone because its conceptually a bad item. I could make a PR to sort out the bad parts (spawning where it shouldn’t, annoying global sound), but I don’t want to because I can’t fix the integral flaws without completely changing the item, or removing as I have elected to do. 

I won’t lie that I hate the item, but that’s because I’m tired of seeing it. It’s annoying and blatant and intrusive. It finds a way of forcing its way into everyones round and pissing them off, all for 7 TC! Buffing the price wouldn’t make sense, since at the end of the day its a niche item without much use at face value, barring one investing their time in other ways. We have all these simple yet clever traitor items available, and then its just “LOL HERE CRAB TAKE CREDITS”. As I said in the PR body, it is at odds with the rest of our items. 

Also I don’t really think we’re constantly changing complaints, we’re just sharing multiple ones we have. I’ve also said this already but don’t interpret the original PRs arguments as my own, because they aren’t. I do understand that the nature of the complaints makes it difficult from a maintainers view point to improve the item, but as you said we just don’t like the item, because its conceptually bad.

**Comment by wesoda25** [comment]
And since everyone has been so eager to mention power sink, I’ll share my thoughts on that now. 

Powersink is interesting and logical. By use of what is effectively an electrical blackhole, a traitor sabotages the power on board the station, freeing them up to do whatever they please. The loss of power enables the traitor, and others such as prisoners or other traitors, to act in a way that they had formerly been barred. It creates interesting situations and is right at home with other uplink items. 

Crab protocol is anything but. Whereas power sink is stealthy and clever, crab protocol is loud, annoying, in your face, and always the same. A global sound, a massive global message, a threat which can’t even be beaten, etc etc. Sure, it provides opportunity to traitors through its beacon like effect. But for what? A mass bombing as imsxz mentioned? Oh yeahh, we should really be enabling that type of gameplay. Its only value is its ability to get people OUT of an area a traitor wants to go into. And, as I’ve been saying from the start, that is entirely too easy for traitors. Crab updates are progressing in a direction in which the entire crew is forced to confront it, hell its invincible now so its basically already there. As economy becomes more and more important, more and more people will come. And a traitor should not have the power to push everyone into an area for 7 TC and the push of a button. Think about it, the only way traitors should be able to exercise that level of control over other players are through large scale events requiring a large degree of luck, skill, knowledge, and preparation. A plasma fire, a released singulo, only those types of events should possess that power!

**Comment by SkeletalElite** [comment]
>Crab updates are progressing in a direction in which the entire crew is forced to confront it, hell its invincible now so its basically already there. As economy becomes more and more important, more and more people will come. And a traitor should not have the power to push everyone into an area for 7 TC and the push of a button. Think about it, the only way traitors should be able to exercise that level of control over other players are through large scale events requiring a large degree of luck, skill, knowledge, and preparation. A plasma fire, a released singulo, only those types of events should possess that power!

You know you can just take our ID, slap it with your ID and then leave and it stops taking you money right. And if you don't give 2 shits about money then you can just ignore it completely. The only people who money really matters to are cargo and miners. I don't think it's as effective at mass bombing as you think. The only people who are really at risk of getting mass bombed are the people who sit there and hit it even though its futile. If you just go to it. click your ID on it and leave the risk of getting caught in a mass bombing or whatever is pretty low unless you are the target of the bomb. 

On top of that as a traitor there's a good chance if you buy this with a specific target in mind, they won't even show up because they dont give a shit about money. You're not "forced" to go deal with it. I really don't see how it having a global announcement / sound effect annoys you, so does a good portion of random events that trigger.

**Comment by wesoda25** [comment]
When I say confront I mean people stopping the credit syphon on their account. And using the current failure of economy to justify the item seems strange to me, since economy is the entire point of it. And since economy will only be improved upon, and make it actually necessary and important, the amount of people who show up shall rise.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Oh yeahh, we should really be enabling that type of gameplay.

Bombs are the issue, not the thing that helps promote an effective bomb use.

You can have an equally damaging bomb without the CRAB.

Do remember you have to go back to your hidey hole so even if people swarm the crab itself, that's not to say there isn't opportunity to steal your target when they're in transit to/from the crab.

**Comment by wesoda25** [comment]
> Bombs are the issue, not the thing that helps promote an effective bomb use.

🤔 

> You can have an equally damaging bomb without the CRAB.

Through careful planning on your own part, as opposed to pressing a button

> Do remember you have to go back to your hidey hole so even if people swarm the crab itself, that's not to say there isn't opportunity to steal your target when they're in transit to/from the crab.

Not really sure what you’re saying here

**Comment by ExcessiveUseOfCobblestone** [comment]
> > Bombs are the issue, not the thing that helps promote an effective bomb use.
> 🤔
> > You can have an equally damaging bomb without the CRAB.
> Through careful planning on your own part, as opposed to pressing a button

Like how you have to carefully plan the timer to coincide with the time most people will be at the crab unit?

The crab unit does NOT guarantee a more effective bomb.

> > Do remember you have to go back to your hidey hole so even if people swarm the crab itself, that's not to say there isn't opportunity to steal your target when they're in transit to/from the crab.
> 
> Not really sure what you’re saying here

Fixed it (maybe?)

**Comment by FantasticFwoosh** [comment]
*cellphone buzzes and rings*
*_Hello Monseuir,_ 
_He PR'd?_

_Duhmp eeit and close"_*

Just hide your station ID credits in a wallet or under a matress or something.

**Comment by ExcessiveUseOfCobblestone** [comment]
I got OOC banned because I commented on the crab phone sound thinking it was a dumb DGL sound so now i'm going to remove it.

**Comment by ExcessiveUseOfCobblestone** [comment]
After looking at it more closely (especially since I got OOC banned from it [STILL MAD]):

I'm not going to remove it and I'm going to keep the punish mechanic of brute forcing it with melee gear.

I will however, entertain the crab being more susceptible to things like guns/bombs/fire. That way if you want to play hero of the station, it's going to take more than just the toolbox.

**Comment by adamkad1** [comment]
well now you get money from it, its good, because money buys things

**Comment by ExcessiveUseOfCobblestone** [comment]
someone can add a text notification in tandem with the sound playing so innocent ~~admins~~ players don't get banned

**Comment by wesoda25** [comment]
Actually lame as fuck

**Comment by wesoda25** [comment]
Also, hero of the station aka bombing then immediately getting banned by admins lol. 

I also still don’t get what you see the benefits of the item are since I still don’t know what you’re saying. If I’m correct in interpreting that you like the item because it provides an opportunity to snag your target going to/from the CRAB, I really don’t think that’s enough to constitute leaving it in. After all thats quite literally mass grief, all for the sake of dragging one person out? Its not subtle, its not clever, its just in your face and annoying. There are so many ways one could use uplink items to the same effect that require actual brain capacity aside from pressing a button. I’m probably just posting in a dead PR here but please do reconsider.

**Comment by ExcessiveUseOfCobblestone** [comment]
if you maxcap a fairly static object that just makes you a dummy. Me saying to make the item more susceptible to these types of damage is certainly not the same as giving permission to use any and all sources of these damages to destroy the object.
